src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1352,10 +1586,16 @@ def get_connected_passed_kwargs(prefix):
                 f"Pipeline {pipeline_class} expected {expected_modules}, but only {passed_modules} were passed."
             )
 
-        # 8. Instantiate the pipeline
+        # 10. Instantiate the pipeline
         model = pipeline_class(**init_kwargs)
 
-        # 9. Save where the model was instantiated from
+        # 11. Handle cases where device map strategy is not "balanced".
+        if device_map == "balanced_ultra_low_memory":
+            model.enable_sequential_cpu_offload(component_device_mapping=final_device_map)
+        elif device_map == "balanced_low_memory":
+            model.enable_model_cpu_offload(component_device_mapping=final_device_map)
```

Comment:
It's the other way around. `balanced_ultra_low_memory` with `enable_model_cpu_offload` and `balanced_low_memory` with `enable_sequential_cpu_offload`

Reply:
I think then there's a misunderstanding. Referring to https://huggingface.co/docs/diffusers/main/en/optimization/memory. 

For `enable_sequential_cpu_offload()`:

> CPU offloading works on submodules rather than whole models. This is the best way to minimize memory consumption, but inference is much slower due to the iterative nature of the diffusion process.

So, this corresponds to the case when we have ultra low memory. Am I missing something?

Reply:
Oh sorry, that's my bad =/ You're correct !